### ISSUE TITLE
"Outdated" with Gnome 40

### ISSUE THREAD

**Issue by somody** [stuartlangridge/gnome-shell-clock-override#57]
https://extensions.gnome.org/extension/1206/clock-override/
https://i.imgur.com/J8GkU4Q.png

**Comment by sluedecke** [comment]
Yup .... this extension does not list 40 as a supported version and some work is needed to get it supported since: ["GNOME Shell 40 now uses GTK4 for extension preferences. This means you need to upgrade your prefs.js UI to GTK4."](https://gjs.guide/extensions/upgrading/gnome-shell-40.html#prefs)

**Comment by sluedecke** [comment]
You can use the attached patch to get it working in Gnome 40 again (although it misses translations).


[support-gnome-40-patch.txt](https://github.com/stuartlangridge/gnome-shell-clock-override/files/6412240/support-gnome-40-patch.txt)

**Comment by wowkina** [comment]
> You can use the attached patch to get it working in Gnome 40 again (although it misses translations).
> 
> [support-gnome-40-patch.txt](https://github.com/stuartlangridge/gnome-shell-clock-override/files/6412240/support-gnome-40-patch.txt)
> 
> or you can use my pull request #59

How to use it this file?

**Comment by sluedecke** [comment]
It is a diff file, most IDE should recognize it. Alternatively you can use the pull request #59.

On May 5, 2021 1:05:11 PM GMT+02:00, Volodymyr Vasylchenko ***@***.***> wrote:
>> You can use the attached patch to get it working in Gnome 40 again (although it misses translations).
>> 
>> [support-gnome-40-patch.txt](https://github.com/stuartlangridge/gnome-shell-clock-override/files/6412240/support-gnome-40-patch.txt)
>> 
>> or you can use my pull request #59
>
>How to use it this file?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/stuartlangridge/gnome-shell-clock-override/issues/57#issuecomment-832601559
-- 
Sascha Lüdecke
+49 179 9210 765

**Comment by mtclare** [comment]
@sluedecke 

I tried your patch.  It seemed to update the files in the correct location:
~/.local/share/gnome-shell/extensions/clock-override@gnomeshell.kryogenix.org/

But after reloading, and even rebooting I am still not able to configure it without errors:
![image](https://user-images.githubusercontent.com/45080773/118321287-66bdaf00-b4cb-11eb-89ee-6377dd728cbe.png)

Is it possible that there is more that needs to be done, besides simply "Make" and "Make Install"?

**Comment by mtclare** [comment]
Update: I copied prefs.js over from your repo and now it works.
